Add controllers

NurievLinar opened the pull request.

NurievLinar committed:
Item. создал поля

NurievLinar committed:
ТЗ 13. Каркас приложения ShareIt.

NurievLinar committed:
ItemRequest.java : удалил неиспользуемые импорты.

sergeymamonov commented on src/main/java/ru/practicum/shareit/item/ItemRepositoryImpl.java:
```diff
@@ -0,0 +1,87 @@
+            return null;
+        }
+        Item item = null;
+        for (Item item1 : itemList) {
```
item1 и item - не очень понятно, какая разница между этими переменными. Нужно использовать слова, чтобы было понятнее. Например, currentItem и так далее.

NurievLinar replied:
Исправил.

sergeymamonov commented on src/main/java/ru/practicum/shareit/item/ItemRepositoryImpl.java:
```diff
@@ -0,0 +1,87 @@
+        List<Item> searchItemsList = new ArrayList<>();
+        for (List<Item> itemList : items.values()) {
+            for (Item item : itemList) {
+                if (item.getAvailable().equals(true) && (item.getName().toLowerCase().contains(substring) ||
```
item.getAvailable().equals(true) - можно проще item.getAvailable()

NurievLinar replied:
Исправил.

sergeymamonov commented on src/main/java/ru/practicum/shareit/user/exception/DuplicateEmailException.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.CONFLICT)
+public class DuplicateEmailException extends Exception {
```
А почему не сделали наследование от RuntimeException? Тогда не нужно будет указывать во всех методах throws DuplicateEmailException

NurievLinar replied:
Исправил. Спасибо.

NurievLinar committed:
ТЗ 13. Исправления приложения ShareIt.
DuplicateEmailException - сделал наследование от RuntimeException;
ItemNotFoundException - сделали наследование от RuntimeException;
UserNotFoundException - сделали наследование от RuntimeException;
В соответствующих методах удалил указывание throws;
ItemRepositoryImpl - исправил item1 на currentItem;
ItemRepositoryImpl - удалил .equals(true);

NurievLinar committed:
Удалил неиспользуемые импорты, мешающие прохождению тестов на GitHub.

NurievLinar committed:
Исправления по Checkstyle, для прохождения тестов на GitHub.
